File: WorkoutPlaner/WorkoutPlaner/WorkoutPlaner.Droid/Services/CalendarService.cs
using Android.Content;
using Android.Provider;
using Java.Util;
using System;
using WorkoutPlaner.Droid.Services;
using WorkoutPlaner.Services;
using Xamarin.Forms;

[assembly: Dependency(typeof(CalendarService))]
namespace WorkoutPlaner.Droid.Services
{
    public class CalendarService : ICalendarService
    {
        public void SaveWorkoutAsync(string workoutName, DateTime timeOfWorkout)
        {
            Intent intent = new Intent(Intent.ActionInsert);
            intent.SetData(CalendarContract.Events.ContentUri);
            intent.PutExtra(CalendarContract.ExtraEventBeginTime,
                GetDateTimeMS(timeOfWorkout.Year, timeOfWorkout.Month, timeOfWorkout.Day, timeOfWorkout.Hour, timeOfWorkout.Minute));
            intent.PutExtra(CalendarContract.ExtraEventBeginTime,
                GetDateTimeMS(timeOfWorkout.Year, timeOfWorkout.Month, timeOfWorkout.Day, timeOfWorkout.Hour + 2, timeOfWorkout.Minute));
            intent.PutExtra(CalendarContract.Events.InterfaceConsts.Title,
                workoutName);
            try
            {
                Forms.Context.StartActivity(intent);
                return;
            }catch(Exception e)
            {
                return;
            }
        }
        long GetDateTimeMS(int yr, int month, int day, int hr, int min)
        {

            Calendar c = Calendar.GetInstance(Java.Util.TimeZone.Default);

            c.Set(Calendar.DayOfMonth, 15);
            c.Set(Calendar.HourOfDay, hr);
            c.Set(Calendar.Minute, min);
            c.Set(Calendar.Month, Calendar.December);
            c.Set(Calendar.Year, 2011);

            return c.TimeInMillis;
        }
    }
}

File: WorkoutPlaner/WorkoutPlaner/app/src/main/java/hu/bme/aut/workoutplaner/service/CalendarService.java
package hu.bme.aut.workoutplaner.service;

import android.content.Context;
import android.content.Intent;
import android.provider.CalendarContract;

import java.util.Calendar;
import java.util.Date;

/**
 * Created by Balazs on 2017. 04. 24..
 */

public class CalendarService {
    private Context context;

    public CalendarService(Context context) {
        this.context = context;
    }
    public void saveEventToCalendar(String title,int year,int month,int day, int hour, int min){
        Calendar calendar = Calendar.getInstance();
        calendar.set(Calendar.YEAR,year);
        calendar.set(Calendar.MONTH, month-1);
        calendar.set(Calendar.DAY_OF_MONTH, day);
        calendar.set(Calendar.HOUR_OF_DAY, hour);
        calendar.set(Calendar.MINUTE, min);
        Date date = new Date(year,month,day,hour,min);
        Intent intent = new Intent(Intent.ACTION_INSERT)
                .setData(CalendarContract.Events.CONTENT_URI)
                .putExtra(CalendarContract.EXTRA_EVENT_BEGIN_TIME, calendar.getTimeInMillis())
                .putExtra(CalendarContract.EXTRA_EVENT_END_TIME, date.getTime()+60*60*1000)
                .putExtra(CalendarContract.Events.TITLE, title);

        context.startActivity(intent);
    }
}
